### PR TITLE
Automate OPERA URLs

### PR DESCRIPTION
**Pull request by jacquelynsmale**

### CODE REVIEW

File: image_services/opera/make_opera_services.py [hunk 1]
```diff
@@ -18,8 +19,17 @@
...
+    _, _, bucket, _, _ = overview_path.split('/')
+    polarization = s3_suffix[1:3]
+    filename = f'{working_directory}/opera_vsis3_{polarization}.csv'
+    key =  f'opera_uris/opera_vsis3_{polarization}.csv'
+
+    s3 = boto3.client('s3')
+    logging.info(f'Downloading {bucket}/{key} to {filename}')
+    url_file = s3.download_file(bucket, key, filename)
+
     with open(url_file, newline='') as urlfile:
```
Comment (asjohnston-asf) [inline image_services/opera/make_opera_services.py:33]: [download_file()](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/download_file.html) doesn't have a return value, so `url_file` ends up as `None` here. You should be able to pass `filename` to the subsequent `open()` call:

```suggestion
    s3.download_file(bucket, key, filename)

    with open(filename, newline='') as urlfile:
```

File: image_services/opera/update_opera_uris.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        response.raise_for_status()
+        for granule in response.json()['items']:
+            for url in granule['umm']['RelatedUrls']:
+                if url['URL'].startswith('s3://') and url['URL'].endswith(f'{polarization}.tif'):
+                    vsis3_uris.append(url['URL'].replace('s3://', '/vsis3/'))
+                    break
+        if 'CMR-Search-After' not in response.headers:
+            break
+        headers['CMR-Search-After'] = response.headers['CMR-Search-After']
+        return vsis3_uris
```
Comment (AndrewPlayer3) [inline image_services/opera/update_opera_uris.py:37]: Is this ```while True:``` loop necessary? The code still seems to work without it.

File: image_services/opera/update_opera_uris.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+
+def upload_file_to_s3(path_to_file: Path, bucket: str, prefix: str = ''):
+    path_to_file = Path(path_to_file)
```
Comment (AndrewPlayer3) [inline image_services/opera/update_opera_uris.py:41]: It feels a little wrong to ask for a Path and then cast to Path right away. Maybe ask for a string and cast to Path, or cast to Path before you use the function.

**Review by AndrewPlayer3 — COMMENTED** [review]
Looks good overall, just a couple nit-pick questions.

File: image_services/opera/update_opera_uris.py (hunk 2 above)
Comment (jacquelynsmale) [reply image_services/opera/update_opera_uris.py:37]: @asjohnston-asf I can't remember what the purpose of that `True` statement is. Was it to handle some sort of CMR outage?

Comment (jacquelynsmale) [reply image_services/opera/update_opera_uris.py]: @AndrewPlayer3 The `while True:` is how we handle multiple pages. I had to double-check with Andrew Johnston about this. MM link with his explanation is here: https://chat.asf.alaska.edu/asf/pl/66kujezwrtg67kd1gcu1mrmtqw

**Review by AndrewPlayer3 — APPROVED** [review]
Everything looks good to me now.

**Pull request merged**